go/pkg/tree/tree.go

```diff
@@ -50,6 +50,36 @@ type Stats struct {
 	// TODO(olaola): number of FileMetadata cache hits/misses go here.
 }
 
+type constructConfig struct {
+	enableSymlink        bool
+	allowDanglingSymlink bool
+}
+
+// ConstructOption controls the behavior when constructing a tree from the local filesystem.
+type ConstructOption = func(c *constructConfig)
+
+// EnableSymlink causes the symlinks to be distinguished from the normal files.
+func EnableSymlink() ConstructOption {
```

Comment:
better to accept bool arg?

Reply:
Ah, since the `tree` package is merged into `client`, I've refactored this part to have a `TreeSymlinkOpts`. PTAL